### PR TITLE
Lps 95418 3rd

### PR DESCRIPTION
**Pull request by arthurchan35**

### CODE REVIEW

File: modules/apps/portal-security-crypto/portal-security-crypto-algorithm-pbkdf2/src/main/java/com/liferay/portal/security/crypto/algorithm/pbkdf2/factory/PBKDF2HashGeneratorFactory.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ */
+@Component(
+	immediate = true, property = "crypto.hash.generator.factory=PBKDF2",
+	service = HashGeneratorFactory.class
+)
+public class PBKDF2HashGeneratorFactory implements HashGeneratorFactory {
+
+	@Override
+	public HashGenerator getGenerator(
+			String algorithmName, JSONObject algorithmMeta)
```
Comment (csierra) [inline modules/apps/portal-security-crypto/portal-security-crypto-algorithm-pbkdf2/src/main/java/com/liferay/portal/security/crypto/algorithm/pbkdf2/factory/PBKDF2HashGeneratorFactory.java]: hey @arthurchan35,

it looks like `algorithName` is not needed for creating the generator, only for getting the proper factory, so the signature for the factory can be just the `algorithmMeta` in all the accepted forms.

Comment (arthurchan35) [reply modules/apps/portal-security-crypto/portal-security-crypto-algorithm-pbkdf2/src/main/java/com/liferay/portal/security/crypto/algorithm/pbkdf2/factory/PBKDF2HashGeneratorFactory.java]: Yes, for this particular PBKDF2HashGenerator, the algorithmName isnt necessary, but for HashGenerator like MessageDigestHashGenerator, an algorithmName is needed. Are you suggesting to hide algorithmName in side algorithmMeta?

**Comment by csierra** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by csierra** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/portal-security-password/portal-security-password-service/src/main/java/com/liferay/portal/security/password/service/impl/HashAlgorithmEntryLocalServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+ *
+ * <p>
+ * This is a local service. Methods of this service will not have security checks based on the propagated JAAS credentials because this service can only be accessed from within the same VM.
+ * </p>
+ *
+ * @author arthurchan35
+ * @see HashAlgorithmEntryLocalServiceBaseImpl
+ */
+@Component(
+	configurationPid = "com.liferay.portal.security.password.configuration.PasswordConfiguration",
```
Comment (csierra) [inline modules/apps/portal-security-password/portal-security-password-service/src/main/java/com/liferay/portal/security/password/service/impl/HashAlgorithmEntryLocalServiceImpl.java]: hey @arthurchan35,
I have been looking around and it looks fine to have `configurationPid` in the localServiceImpl... we already have examples in other modules. 

Thx a ton!

**Comment by arthurchan35** [comment]
ci:test

**Comment by arthurchan35** [comment]
ci:test

**Comment by arthurchan35** [comment]
ci:test:relevant

**Comment by arthurchan35** [comment]
ci:test:relevant

**Comment by arthurchan35** [comment]
ci:test:relevant

**Comment by arthurchan35** [comment]
ci:test:sf

**Comment by arthurchan35** [comment]
ci:stop

**Comment by arthurchan35** [comment]
ci:test:relevant

**Pull request closed without merging**